plugin/storage/es/spanstore/reader.go

```diff
@@ -235,6 +235,9 @@ func (s *SpanReader) multiRead(traceIDs []string, startTime, endTime time.Time)
 		if result.Hits == nil {
 			return nil, errNilHits
 		}
+		if len(result.Hits.Hits) == 0 {
```

Comment:
* under what conditions will this happen? (maybe add a comment)
  * why should this be handled differently than `result.Hits == nil`?

Reply:
when trace id is invalid, we got response json as follows:
```json
{
   "responses":[
      {
         "took":2,
         "timed_out":false,
         "_shards":{
            "total":15,
            "successful":15,
            "failed":0
         },
         "hits":{
            "total":0,
            "max_score":null,
            "hits":[]
         },
         "status":200
      }
   ]
}
```
and then the response json is unmarshalled to the `elastic.MultiSearchResult` struct, you will notice that the snippet
```json
"hits":{
    "total":0,
    "max_score":null,
    "hits":[]
},
```
will not get nil `result.Hits`

Reply:
What if we combine both `if`s into this instead:
```
if result.Hits == nil || len(result.Hits.Hits) == 0 {
    continue
}
```
I think this will work better with the calling function that will detect empty cumulative result and respond with ErrNotFound. The private `errNilHits` should be removed altogether.

Reply:
OK, I will do it